spec/filters/split_spec.rb

```diff
@@ -82,8 +82,38 @@
     end
 
     sample("array" => ["single-element"], "untouched" => "1\n2\n3") do
-      insist { subject["array"] } == "single-element"
-      insist { subject["untouched"] } == "1\n2\n3"
+      insist { subject.get("array") } == "single-element"
+      insist { subject.get("untouched") } == "1\n2\n3"
+    end
+  end
+
+  describe "split array with string interpolation" do
+    config <<-CONFIG
+      filter {
+        split {
+          field => "%{array_field}"
+        }
+      }
+    CONFIG
+
+    sample("array" => ["big", "bird", "sesame street"], "array_field" => "array", "untouched" => "1\n2\n3") do
+      insist { subject.length } == 3
+      subject.each do |s|
+         insist { s.get("untouched") } == "1\n2\n3"
+      end
+      insist { subject[0].get("array") } == "big"
+      insist { subject[1].get("array") } == "bird"
+      insist { subject[2].get("array") } == "sesame street"
+    end
+
+    sample("array" => ["big"], "array_field" => "array", "untouched" => "1\n2\n3") do
+      insist { subject.is_a?(Logstash::Event) }
+      insist { subject.get("array") } == "big"
+    end
+
+    sample("array" => ["single-element"], "array_field" => "array", "untouched" => "1\n2\n3") do
+      insist { subject.get("array") } == "single-element"
+      insist { subject.get("untouched") } == "1\n2\n3"
```

Comment:
Isn't this last test the same as the preceding one? I don't get why it's been split in two.